src/libsyntax_ext/assert.rs

```diff
@@ -86,5 +86,12 @@ fn parse_assert<'a>(
         } else {
             None
         },
-    })
+    };
+
+    if parser.token != token::Eof {
+        parser.expect_one_of(&[], &[])?;
```

Comment:
You can add a check here for `parser.token == token::Semi` and swallow it or warn about it, depending on which way we want to go about it.

You can also try parser.parse_tokens() and modify `assert.custom_message` with the result, and complain about it then.